### PR TITLE
fix comparison used to determine change in invalidations

### PR DESCRIPTION
**Pull request by KristofferC**
https://github.com/julia-actions/julia-invalidations/pull/4 made the comparison go from a number comparison to a string comparison effectively breaking the workflow.

Fixes https://github.com/julia-actions/julia-invalidations/issues/16, https://github.com/julia-actions/julia-invalidations/issues/8

### CODE REVIEW

**Comment by KristofferC** [comment]
Is it really intended to install this workflow by copy pasting the check into every repo here. A bug in that code (which is the case here) means that every single repo has to update their workflow. Shouldn't this be versioned somehow?

Anyway, there should probably be some mass action that opens PR to fix all the places where this workflow got added.

**Comment by KristofferC** [comment]
It might be possible to fix the existing actions by being sufficiently clever in updating how these values are stored:

https://github.com/julia-actions/julia-invalidations/blob/5d237091b8285114de7b3136f634b3bbc0dc35e9/action.yml#L10-L16

so that if `a < b` then `string(a) < string(b)`.

**Comment by ranocha** [comment]
Would changing

https://github.com/julia-actions/julia-invalidations/blob/5d237091b8285114de7b3136f634b3bbc0dc35e9/action.yml#L61-L62

to something like

```julia
           println(io, @sprintf("total=%09d", inv_total)")
           println(io, @sprintf("deps=%09d", inv_deps)")
```

do this trick?

**Comment by KristofferC** [comment]
Maybe, I'm not sure exactly how github treats these things so I think it just has to be tested in practice.

**Comment by IanButterworth** [comment]
Has that been tried? It'd be good to get this in given maintenance attention on the action right now.

File: README.md [hunk 1]
```diff
@@ -54,8 +54,10 @@ jobs:
         echo "Invalidations on default branch: ${{ steps.invs_default.outputs.total }} (${{ steps.invs_default.outputs.deps }} via deps)" >> $GITHUB_STEP_SUMMARY
         echo "This branch: ${{ steps.invs_pr.outputs.total }} (${{ steps.invs_pr.outputs.deps }} via deps)" >> $GITHUB_STEP_SUMMARY
     - name: Check if the PR does increase number of invalidations
-      if: steps.invs_pr.outputs.total > steps.invs_default.outputs.total
-      run: exit 1
+      run: |
+        if [ ${{ steps.invs_pr.outputs.total }} -gt ${{ steps.invs_default.outputs.total }} ]; then
+          exit 1
+        fi
```
Comment (IanButterworth) [inline README.md:60]: To keep the logic at whether the step runs or not
```suggestion
      if: ${{ fromJSON(steps.invs_pr.outputs.total) > fromJSON(steps.invs_default.outputs.total) }}
      run: exit 1
```

**Comment by IanButterworth** [comment]
With https://github.com/julia-actions/julia-invalidations/pull/35 in and released, I don't think this is needed.

**Comment by IanButterworth** [comment]
Actually I think this protects against the case where there are more than 1e9 invalidations, so still worth doing

**Pull request merged**